### PR TITLE
Update Input Validation

### PR DESCRIPTION
**Pull request by otkd**
- Clarify Server-side & Client-side recommendations
- Add references
- Style changes for uniformity

Thank you for submitting a Pull Request (PR) to the Cheat Sheet Series. 

> :triangular_flag_on_post: If your PR is related to grammar/typo mistakes, please double-check the file for other mistakes in order to fix all the issues in the current cheat sheet.

Please make sure that for your contribution:

- [x] In case of a new Cheat Sheet, you have used the [Cheat Sheet template](https://github.com/OWASP/CheatSheetSeries/blob/master/templates/New_CheatSheet.md).
- [x] All the markdown files do not raise any validation policy violation, see the [policy](https://github.com/OWASP/CheatSheetSeries/actions?query=workflow%3A%22Markdown+Link+Check%22).
- [x] All the markdown files follow these [format rules](https://github.com/OWASP/CheatSheetSeries/blob/master/CONTRIBUTING.md#markdown).
- [x] All your assets are stored in the **assets** folder.
- [x] All the images used are in the **PNG** format.
- [x] Any references to websites have been formatted as `[TEXT](URL)`
- [x] You verified/tested the effectiveness of your contribution (e.g., the defensive code proposed is really an effective remediation? Please verify it works!).
- [x] The CI build of your PR pass, see the build status [here](https://github.com/OWASP/CheatSheetSeries/actions).

If your PR is related to an issue, please finish your PR text with the following line:

This PR covers issue #<insert number here>.

Thank you again for your contribution :smiley:

### CODE REVIEW

File: cheatsheets/Input_Validation_Cheat_Sheet.md [hunk 1]
```diff
@@ -119,13 +118,13 @@ Some Allowlist validators have also been predefined in various open source packa
 
 - [Apache Commons Validator](http://commons.apache.org/proper/commons-validator/)
 
-## Client Side vs Server Side Validation
+## Client-side vs Server-side Validation
 
-Be aware that any JavaScript input validation performed on the client can be bypassed by an attacker that disables JavaScript or uses a Web Proxy. Ensure that any input validation performed on the client is also performed on the server.
+Input validation **must** be implemented on the server-side before any data is processed by an application’s functions, as any JavaScript-based input validation performed on the client-side can be circumvented by an attacker who disables JavaScript or uses a web proxy. Implementing both client-side JavaScript-based validation for UX and server-side validation for security is the recommended approach, leveraging each for their respective strengths
```
Comment (szh) [inline cheatsheets/Input_Validation_Cheat_Sheet.md:123]:
```suggestion
Input validation **must** be implemented on the server-side before any data is processed by an application’s functions, as any JavaScript-based input validation performed on the client-side can be circumvented by an attacker who disables JavaScript or uses a web proxy. Implementing both client-side JavaScript-based validation for UX and server-side validation for security is the recommended approach, leveraging each for their respective strengths.
```
Missing period.

**Review by szh — DISMISSED** [review]

**Review by kwwall — CHANGES_REQUESTED** [review]
My $.02: "Whitelist" was in the dictionary (e.g., see https://dictionary.cambridge.org/us/dictionary/english/whitelist). But while "allowlist" is it a lot of technical glossaries and is broadly used, I haven't yet found it in any prominent  dictionary of the English language (of any variety). So, until it is--or if you can convince the Cambridge Dictionary, the American Heritage Dictionary, etc. to add it--I think we should continue to write it out as 2 hyphenated words, e.g., "allow-list". (Same with "block-list" if you changed that to "blocklist".) Other than that, LGTM.

**Comment by otkd** [comment]
> My $.02: "Whitelist" was in the dictionary (e.g., see https://dictionary.cambridge.org/us/dictionary/english/whitelist). But while "allowlist" is it a lot of technical glossaries and is broadly used, I haven't yet found it in any prominent dictionary of the English language (of any variety). So, until it is--or if you can convince the Cambridge Dictionary, the American Heritage Dictionary, etc. to add it--I think we should continue to write it out as 2 hyphenated words, e.g., "allow-list". (Same with "block-list" if you changed that to "blocklist".) Other than that, LGTM.

This perhaps would be a reasonble approach if all terminology used by OWASP would be in general purpose dictionaries, however given the intended audience this is unlikely to be the case.

"Allowlist" is for example recommended by:
- [NIST](https://csrc.nist.gov/glossary/term/whitelist)
- [Inclusive Naming Initiative](https://inclusivenaming.org/word-lists/tier-1/whitelist/) members: Cisco, Intel, Red Hat, Canonical, CNCF, Extreme Networks etc.
- [Google](https://developers.google.com/style/inclusive-documentation#replace)
- [Linux Kernel](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/Documentation/process/coding-style.rst?id=a5f526ecb075a08c4a082355020166c7fe13ae27)

File: cheatsheets/Input_Validation_Cheat_Sheet.md [hunk 2]
```diff
@@ -54,8 +53,8 @@ It's also free-form text input that highlights the importance of proper context-
 The primary means of input validation for free-form text input should be:
 
 - **Normalization:** Ensure canonical encoding is used across all the text and no invalid characters are present.
-- **Character category allow-listing:** Unicode allows listing categories such as "decimal digits" or "letters" which not only covers the Latin alphabet but also various other scripts used globally (e.g. Arabic, Cyrillic, CJK ideographs etc).
-- **Individual character allow-listing:** If you allow letters and ideographs in names and also want to allow apostrophe `'` for Irish names, but don't want to allow the whole punctuation category.
+- **Character category allowlisting:** Unicode allows listing categories such as "decimal digits" or "letters" which not only covers the Latin alphabet but also various other scripts used globally (e.g. Arabic, Cyrillic, CJK ideographs etc).
```
Comment (kwwall) [inline cheatsheets/Input_Validation_Cheat_Sheet.md:56]: In https://github.com/OWASP/CheatSheetSeries/pull/1429#issuecomment-2171935485, @otkd wrote:
> This perhaps would be a reasonble approach if all terminology used by OWASP would be in general purpose dictionaries, however given the intended audience this is unlikely to be the case.
> 
> "Allowlist" is for example recommended by:
> 
>     NIST
>     Inclusive Naming Initiative members: Cisco, Intel, Red Hat, Canonical, CNCF, Extreme Networks etc.
>     Google
>     Linux Kernel

My concern is not that the audience of the CS will fail to understand it. It's more of a concern that until it becomes part of the official dictionaries, it's going to get flagged by spell-checkers, which is just annoying as hell. (For instance, if Firefox in US_en, "allowlist" has the little red squiggle, despite me adding to a custom dictionary numerous times.)

**Review by kwwall — APPROVED** [review]
This is not a showstopper. I'm not going to allow the perfect to become the enemy of the good here. But it wouldn't surprise me to see someone create an issue and a PR later on claiming it's not in the dictionary. (Although, that someone will not be me.)

**Comment by jmanico** [comment]
"allowlist" is the NIST standard. Dictionaries be dammed. Let's go with allowlist. 

Other dictionaries that support this:

- https://en.wiktionary.org/wiki/allowlist
- https://www.collinsdictionary.com/submission/407714/allowlist
- https://compliancedictionary.com/term/384420

**Review by jmanico — APPROVED** [review]

**Pull request merged**